generator/templates/go/go.jinja2

```diff
@@ -0,0 +1,723 @@
+/////////////////////////////////////////////////////////////////////////
```

Comment:
I have reviewed only tests right now, as this takes a while. As _most_ of go code is hidden inside generator, I think this is one of the most important files. Thus, please split this file into multiple ones. You can use jinja marcos to extract functions. This will help a lot to read and understand the generated code and template itself.